lightfm/lightfm.py

```diff
@@ -732,8 +732,19 @@ def predict(self, user_ids, item_ids, item_features=None,
 
         return predictions
 
+    def _check_test_train_intersections(self, test_mat, train_mat):
+        if train_mat is not None:
+            bool_intersections_mat = test_mat.astype(bool).multiply(train_mat.astype(bool))
+            n_intersections = np.sum(bool_intersections_mat.tocsr().data)
```

Comment:
Can we not just check `n_intersections = bool_intersections.getnnz()`?

Reply:
right again.